src/libPOW/pow.cpp

```diff
@@ -474,21 +476,23 @@ void POW::InitCUDA() {
 #ifdef CUDA_MINE
   using namespace dev::eth;
 
-  if (CUDAMiner::getNumDevices() < NUM_DEVICE_TO_USE) {
-    LOG_GENERAL(FATAL, "NUM_DEVICE_TO_USE "
-                           << NUM_DEVICE_TO_USE
-                           << " is more than the physical CUDA GPU number "
-                           << CUDAMiner::getNumDevices());
-  }
-
   if (!CUDAMiner::configureGPU(CUDA_BLOCK_SIZE, CUDA_GRID_SIZE, CUDA_STREAM_NUM,
                                CUDA_SCHEDULE_FLAG, 0, 0, false, false)) {
     LOG_GENERAL(FATAL, "Failed to configure CUDA GPU, please check hardware");
   }
 
   CUDAMiner::setNumInstances(UINT_MAX);
-  for (uint32_t i = 0; i < NUM_DEVICE_TO_USE; ++i) {
-    m_miners.push_back(std::make_unique<CUDAMiner>(i));
+  auto setGpuIndex = GetGpuToUse();
```

Comment:
The naming of variable should try to keep away of what the implementation it. Then, if someday we need to modify the implementation, we don't need to change the variable name again.